source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetResourceDelegate.h

```diff
@@ -0,0 +1,21 @@
+//
+//  ACRChoiceSetResourceDelegate.h
+//  AdaptiveCards
+//
+//  Created by Ankit Maurya on 8/22/18.
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "ACOAdaptiveCard.h"
+#import "ACOBaseActionElement.h"
+
+@protocol ACRChoiceSetResourceDelegate <NSObject>
```

Comment:
I suggest to use ACOResourceResolvers
1. create ACOResourceResolvers, and register custom ACOResourceResolver with scheme.
Scheme can be any NSString
 [customResolvers setResourceResolver:customRsolver scheme: @"ACRChoiseSetResorceResolverSchme"];

2. implement ACOIResourceResolver interface;

3. to retrieve image, add code that query the resourceResolvers with the scheme, and get image using the interface.